controller/engine_controller.go

```diff
@@ -457,6 +457,13 @@ func (ec *EngineController) CreateInstance(obj interface{}) (*longhorn.InstanceP
 		return nil, err
 	}
 
+	if e.Status.InstanceManagerName == "" {
+		e.Status.InstanceManagerName = im.Name
+	}
+	if e.Status.InstanceManagerName != im.Name {
+		return nil, fmt.Errorf("found instance manager name conflict %s vs %s during engine instance creation", e.Status.InstanceManagerName, im.Name)
+	}
```

Comment:
If we hit the issue, how to work around it? Force detaches the volume?

Reply:
Yes. At least we should not continue the attachment